be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -60,6 +60,9 @@ Status SegmentWriter::init(uint32_t write_mbytes_per_sec) {
         DCHECK(type_info != nullptr);
 
         ColumnWriterOptions opts;
+        if (column.type() == OLAP_FIELD_TYPE_CHAR || column.type() == OLAP_FIELD_TYPE_VARCHAR) {
+            opts.encoding_type = DICT_ENCODING;
```

Comment:
You don't need to explicitly set encoding_type here because the default encoding for char and varchar is DICT_ENCODING.

Reply:
👌

Reply:
forgot to remove?

Reply:
content rollback when merging branch